### ISSUE TITLE
Replace bin/cluster-preflight-check with autogenerated script from setup.py

### ISSUE THREAD

**Issue by tomschr** [liangxin1300/cluster_preflight_check#9]
## Problem

You don't need the script in `bin/cluster-preflight-check`. The setuptools allows you to auto generate the script, be it for Python2 or 3.

## Benefits

* You don't need a separate, hand-crafted script
* It's done all automatically through `setup.py`
* It's Python standard packaging practice
* The combination of Python commands and shell script looks potentially insecure to me; we shouldn't reinvent the wheels again if we can avoid that with a standard method. :wink:

Cons:

* You need a virtual environment to generate the script
   (I don't consider it as a real disadvantage as you need a virtual environment anyway for your testing).

## Expected Solution

In a nutshell, pass the `entry_points` parameter to the `setuptools.setup` function:

* Remove the file `bin/cluster-preflight-check`.
* Remove in `setup.py` the following line:

    ```python
    scripts=['bin/cluster-preflight-check'],
    ```

* Add another function `main()` in your `cluster_preflight_check/main.py` file which the following content:

    ```python
    def main():
        ctx = Context()
        run(ctx)
    ```

* Add the following lines into your `setup.py`.

    ```python
    entry_points={
        'console_scripts': ['cluster-preflight-check=cluster_preflight_check.main:main'],
    }
    ```

**Comment by liangxin1300** [comment]
Thanks!
